### DISCUSSION TITLE
Error building modelserver project template

### DISCUSSION THREAD

**Discussion by kafram** [eclipse-emfcloud/emfcloud#201] [ModelServer] [answered]
Hi all! 

I'm trying to build the project template modelserver-glspjava-emf-theia on linux debian-base, but maven has reported an error "filename too long", as:

[ERROR] Failed to execute goal com.googlecode.addjars-maven-plugin:addjars-maven-plugin:1.0.5:add-jars (default) on project org.eclipse.emfcloud.integration.example.modelserver: Failed to install artifact org.eclipse.emfcloud.modelserver.glsp:org.eclipse.emfcloud.integration.example.modelserver-org.eclipse.emfcloud.modelserver.glsp.notation.commands-0.7.0-SNAPSHOT.jar:jar:0.7.0-SNAPSHOT: /home/carlos/.m2/repository/org/eclipse/emfcloud/modelserver/glsp/org.eclipse.emfcloud.integration.example.modelserver-org.eclipse.emfcloud.modelserver.glsp.notation.commands-0.7.0-SNAPSHOT.jar/0.7.0-SNAPSHOT/org.eclipse.emfcloud.integration.example.modelserver-org.eclipse.emfcloud.modelserver.glsp.notation.commands-0.7.0-SNAPSHOT.jar-0.7.0-SNAPSHOT.jar (Nome de arquivo muito longo) -> [Help 1] 

Searching for a solution, I found that this limitation (255 chars on filename) is on linux filesystem. 

Does anyone know what I can do to fix this?
 
Best regards,

Carlos.

**Comment by ndoschek** [comment, marked as answer]
Hi @kafram!

Thank you for interest in EMF cloud!

Until now we did not face issues with linux systems, we use Ubuntu ourselves for the most part.
Which Java and Maven versions are you using?

BR,
Nina

**Reply by kafram** [reply]
Hi @ndoschek 
Thank you for your help.

I'm using Linux Mint 21.1, OpenJDK 11, and Maven 3.6.3.

Best regards,
Carlos.

**Reply by xai** [reply]
Hi @kafram!

On which filesystem are you building the project? The filename mentioned in the above error message is 147 characters long, which should not be a problem with, e.g., ext4, which indeed has a 255-character limit for file names.

What is the output of `getconf NAME_MAX .` and `getconf PATH_MAX .` in the directory where you build the project?

Kind regards,
Olaf

**Reply by kafram** [reply]
Hi @xai 
Thank you for your help!

I've tried what you suggested, and I realized that on the root folder, the max name chars were 255, but inside the folder was 143. So I Googled it and found that it happens when an encrypted filesystem is enabled on the user folder.
To be registered if someone needs it in the future, I'll fix it by creating another user account without an encrypted filesystem, or moving the installation project to another folder outside the home.

Best regards,
Carlos.
